libraries/Microsoft.Bot.Builder.Expressions/CommonRegex.cs

```diff
@@ -1,24 +1,59 @@
 ﻿using System;
 using System.Collections.Generic;
-using System.Text;
+using System.Linq;
 using System.Text.RegularExpressions;
 using Antlr4.Runtime;
 using Antlr4.Runtime.Misc;
 using Antlr4.Runtime.Tree;
 
 namespace Microsoft.Bot.Builder.Expressions
 {
+    internal class RegexLRUEntity
+    {
+        public RegexLRUEntity(long timestamp, Regex regex)
+        {
+            this.Timestamp = timestamp;
+            this.Regex = regex;
+        }
+
+        /// <summary>
+        /// Gets time when this item inserted.
+        /// </summary>
+        /// <value>
+        /// Time when this item inserted.
+        /// </value>
+        public long Timestamp { get; }
+
+        /// <summary>
+        /// Gets regex entity of cache.
+        /// </summary>
+        /// <value>
+        /// Regex entity of cache.
+        /// </value>
+        public Regex Regex { get; }
+    }
+
     public class CommonRegex
     {
+        private const int CacheCapacity = 15;
```

Comment:
We should be able to add\remove cache almost transparent to the caller code. so we should create a LRUCache class maybe to do this, instead of exposing too much implementation details in this layer